server/filestore.go

```diff
@@ -2273,7 +2274,7 @@ func (mb *msgBlock) firstMatchingMulti(sl *Sublist, start uint64, sm *StoreMsg)
 		}
 		expireOk := seq == lseq && mb.llseq == seq
 
-		if r := sl.Match(fsm.subj); len(r.psubs) > 0 {
+		if r := sl.MatchInline(fsm.subj, &result); len(r.psubs) > 0 {
```

Comment:
I think MatchWithResult would be better here as a name.

Reply:
That's a much better name, updated.